devops/generate_systemd.py

```diff
@@ -0,0 +1,105 @@
+
+    # dbserver
+    dbserver = f'/usr/sbin/arangod -c {pwd}/dbserver8530/arangod.conf --database.directory {pwd}/dbserver8530/data --javascript.startup-directory /usr/share/arangodb3/js --javascript.app-path {pwd}/dbserver8530/apps --log.file {pwd}/dbserver8530/arangod.log --log.force-direct false --server.jwt-secret-keyfile {pwd}/dbserver8530/arangod.jwtsecret --javascript.copy-installation true --cluster.my-address tcp://{my_ip}:8530 --cluster.my-role PRIMARY --foxx.queues false --server.statistics true {cluster_agency_endpoints}'
+
+    # coordinator
+    coordinator = f'/usr/sbin/arangod -c {pwd}/coordinator8529/arangod.conf --database.directory {pwd}/coordinator8529/data --javascript.startup-directory /usr/share/arangodb3/js --javascript.app-path {pwd}/coordinator8529/apps --log.file {pwd}/coordinator8529/arangod.log --log.force-direct false --server.jwt-secret-keyfile {pwd}/coordinator8529/arangod.jwtsecret --javascript.copy-installation true --cluster.my-address tcp://{my_ip}:8529 --cluster.my-role COORDINATOR --foxx.queues true --server.statistics true {cluster_agency_endpoints}'
+
+    return agency, dbserver, coordinator
+
+
+def generate_systemd(node_number, dry_run=False):
+    agency, dbserver, coordinator = get_cmds(node_number)
+
+    if dry_run:
+        print(cmd.format(arango_cmd=dbserver, name='dbserver',
+              description='DBserver process', pwd=path))
+        print(cmd.format(arango_cmd=coordinator, name='coordinator',
+              description='Coordinator process', pwd=path))
+        if node_number <= 3:
+            print(cmd.format(arango_cmd=agency, name='agency',
+                  description='Agency process', pwd=path))
+    else:
+        with open('./arango_dbserver.service', 'w') as dbserver_systemd:
+            dbserver_systemd.write(cmd.format(
+                arango_cmd=dbserver, name='dbserver', description='DBserver process', pwd=path))
+
+        with open('./arango_coordinator.service', 'w') as coordinator_systemd:
+            coordinator_systemd.write(cmd.format(
+                arango_cmd=coordinator, name='coordinator', description='Coordinator process', pwd=path))
+
+        if node_number <= 3:
+            with open('./arango_agency.service', 'w') as agency_systemd:
+                agency_systemd.write(cmd.format(
+                    arango_cmd=agency, name='agency', description='Agency process', pwd=path))
+
+
+if __name__ == '__main__':
+    # Get the number of input arguments.
+    num_args = len(sys.argv)
+
+    if num_args < 1:
+        print('Node number required (1, 2, 3, or 4). Usage: $ python3 generate_systemd.py <node_number>')
+        exit(1)
+
+    try:
+        node_number = int(sys.argv[1])
```

Comment:
should also check the value of the node number to make sure it is in [1, 2, 3, 4].

Reply:
good catch! just added